### PR TITLE
Feat: add "from" and "to" query params to GET /games

### PR DESCRIPTION
**Pull request by maxijabase**

### CODE REVIEW

File: src/shared/pipes/parse-date.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { BadRequestException, Injectable, PipeTransform } from '@nestjs/common';
+import moment = require('moment');
+
+@Injectable()
+export class ParseDatePipe implements PipeTransform {
+  transform(value: string): Date | undefined {
+    if (!value) {
+      return undefined;
+    }
+    const date = moment(value, 'YYYY-MM-DD', true);
```
Comment (garrappachc) [inline src/shared/pipes/parse-date.pipe.ts]: Use `date-fns` instead of `moment`

**Review by garrappachc — CHANGES_REQUESTED** [review]

**Comment by garrappachc** [comment]
Nice one, thanks!

**Comment by maxijabase** [comment]
The functionality I intended is to check against the `ended` date, since that would mean a game actually finished and can count as "valid"? Or since that's what I want to do, is it best to check for the `created` date and pass `state=ended`?

Also, is it OK the way `limit` is currently being handled with the introduction of this filter? Is it intuitive to make the player guess the amount of games that will be within the specified timestamp, or make him pass `limit=0` to bring all results?

**Comment by garrappachc** [comment]
> The functionality I intended is to check against the `ended` date, since that would mean a game actually finished and can count as "valid"? Or since that's what I want to do, is it best to check for the `created` date and pass `state=ended`?

`game.state` will be either `ended` or `interrupted`, but the `ended` event will be always present for ended games. I think I'd be better to compare the `created.at` instead of `ended`, since the user can also pass the `state` filter to the query and achieve the desired result, e.g.:
`/games?from=2023-01-01&to=2024-01-01&state=ended,interrupted`

> Also, is it OK the way `limit` is currently being handled with the introduction of this filter? Is it intuitive to make the player guess the amount of games that will be within the specified timestamp, or make him pass `limit=0` to bring all results?

The `limit` param is always present; the number of results is available in the `itemCount` field in the response.

**Review by garrappachc — APPROVED** [review]

**Pull request merged**